recipes/mpg123/all/conanfile.py

```diff
@@ -43,153 +50,162 @@ class Mpg123Conan(ConanFile):
         "seektable": "1000",
         "module": "dummy",
     }
-    exports_sources = "CMakeLists.txt", "patches/**"
-    generators = "cmake", "pkg_config", "cmake_find_package"
 
-    _autotools = None
-    _cmake = None
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _audio_module(self):
+        return {
+            "libalsa": "alsa",
+        }.get(str(self.options.module), str(self.options.module))
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        if is_msvc(self):
+            cmake_layout(self, src_folder="src")
+        else:
+            basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.module == "libalsa":
-            self.requires("libalsa/1.2.4")
+            self.requires("libalsa/1.2.7.2")
         if self.options.module == "tinyalsa":
-            self.requires("tinyalsa/1.1.1")
+            self.requires("tinyalsa/2.0.0")
 
     def validate(self):
-        try:
-            int(self.options.seektable)
-        except ValueError:
-            raise ConanInvalidConfiguration("seektable must be an integer")
-        if self.settings.os != "Windows":
-            if self.options.module == "win32":
-                raise ConanInvalidConfiguration("win32 is an invalid module for non-Windows os'es")
+        if not str(self.options.seektable).isdigit():
+            raise ConanInvalidConfiguration(f"The option -o {self.ref.name}:seektable must be an integer number.")
+        if self.settings.os != "Windows" and self.options.module == "win32":
+            raise ConanInvalidConfiguration(f"The option -o {self.ref.name}:module should not use 'win32' for non-Windows OS")
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        self.build_requires("pkgconf/1.7.4")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
```

Comment:
Why are we adding such switches again? In the past, we had the opinion that such switches only introduce trouble

Reply:
configuration does not affect the package, so users can add or not without affect the resulting package. But on Conan 2.x, tool_requires affect the package ID, so we can reproduce more cases, or at least, know when users are using or not build requirements.

Reply:
I would still prefer to control the tool_requires globally